pallets/schemas/src/rpc/src/lib.rs

```diff
@@ -25,20 +55,57 @@ impl<C, M> SchemasHandler<C, M> {
 	}
 }
 
-impl<C, Block, AccountId> SchemasApi<<Block as BlockT>::Hash, AccountId>
+impl<C, Block, AccountId, Balance>
+	SchemasApi<<Block as BlockT>::Hash, AccountId, RuntimeDispatchInfo<Balance>>
 	for SchemasHandler<C, Block>
 where
 	Block: BlockT,
 	C: 'static + ProvideRuntimeApi<Block> + HeaderBackend<Block>,
-	C::Api: SchemasRuntimeApi<Block, AccountId>,
+	C::Api: SchemasRuntimeApi<Block, AccountId, Balance>,
 	AccountId: Codec,
+	Balance: Codec + MaybeDisplay + Copy + TryInto<NumberOrHex>,
 {
 	fn get_latest_schema_id(&self, at: Option<<Block as BlockT>::Hash>) -> Result<u16> {
 		let api = self.client.runtime_api();
 		let at = BlockId::hash(at.unwrap_or_else(|| self.client.info().best_hash));
 		let schema_api_result = api.get_latest_schema_id(&at);
 		map_result(schema_api_result)
 	}
+
+	fn get_schema_registration_fee(
+		&self,
+		at: Option<<Block as BlockT>::Hash>,
+		schema: Vec<u8>,
+	) -> Result<FeeDetails<NumberOrHex>> {
+		let api = self.client.runtime_api();
+		let at = BlockId::hash(at.unwrap_or_else(|| self.client.info().best_hash));
+		let fee_details = api.calculate_schema_cost(&at, schema).map_err(|e| RpcError {
+			code: ErrorCode::ServerError(Error::RuntimeError.into()),
+			message: "Unable to calculate schema fee.".into(),
+			data: Some(e.to_string().into()),
+		})?;
+
+		let try_into_rpc_balance = |value: Balance| {
+			value.try_into().map_err(|_| RpcError {
+				code: ErrorCode::InvalidParams,
+				message: format!("{} doesn't fit in NumberOrHex representation", value),
+				data: None,
+			})
+		};
+
+		Ok(FeeDetails {
+			inclusion_fee: if let Some(inclusion_fee) = fee_details.inclusion_fee {
+				Some(InclusionFee {
+					base_fee: try_into_rpc_balance(inclusion_fee.base_fee)?,
+					len_fee: try_into_rpc_balance(inclusion_fee.len_fee)?,
+					adjusted_weight_fee: try_into_rpc_balance(inclusion_fee.adjusted_weight_fee)?,
+				})
+			} else {
+				None
+			},
+			tip: Default::default(),
+		})
+	}
```

Comment:
Gets the schema cost from calculate_schema_cost implemented in Schemas pallet and translate into  return type